metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -60,12 +62,14 @@ def create_instrument(kind, name, unit, description, callback)
         raise InstrumentNameError if name.empty?
         raise InstrumentNameError unless NAME_REGEX.match?(name)
         raise InstrumentUnitError if unit && (!unit.ascii_only? || unit.size > 63)
-        raise InstrumentDescriptionError if description && (description.size > 1023 || !utf8mb3_encoding?(description))
+        raise InstrumentDescriptionError if description && (description.size > 1023 || !utf8mb3_encoding?(description.dup))
 
         @mutex.synchronize do
-          raise DuplicateInstrumentError if @registry.include? name
+          if @instrument_registry.include? name
+            OpenTelemetry.logger.warn("duplicate instrument registration occured for instrument #{name}")
```

Comment:
```suggestion
            OpenTelemetry.logger.warn("duplicate instrument registration occurred for instrument #{name}")
```

Do we want to proceed with the registration in this case? I think we'd want to keep the existing instrument and ignore the duplicate one.

Reply:
I probably missed the change and its associated discussion, but why do we want to log rather than raising here?

Reply:
I should not have brought this into this PR, but the API spec changed.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-type-conflict-detection

>When more than one Instrument of the same name is created for identical Meters, denoted duplicate instrument registration, the implementation MUST create a valid Instrument in every case. Here, "valid" means an instrument that is functional and can be expected to export data, despite potentially creating a [semantic error in the data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#opentelemetry-protocol-data-model-producer-recommendations).

>It is unspecified whether or under which conditions the same or different Instrument instance will be returned as a result of duplicate instrument registration. The term identical applied to Instruments describes instances where all identifying fields are equal. The term distinct applied to Instruments describes instances where at least one field value is different.

>When more than one distinct Instrument is registered with the same name for identical Meters, the implementation SHOULD emit a warning to the user informing them of duplicate registration conflict(s). The warning helps to avoid the semantic error state described in the [OpenTelemetry Metrics data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#opentelemetry-protocol-data-model-producer-recommendations) when more than one Metric is written for a given instrument name and Meter identity by the same MeterProvider.